azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/SetupControlBarViewModel.kt

```
@@ -153,12 +179,32 @@ internal class SetupControlBarViewModel(
     ): Boolean {
         return !shouldControlsBeDisabled(callingState) &&
             audioStateOperation != AudioOperationalStatus.PENDING &&
-            setupScreenOptions?.isMicrophoneButtonEnabled != false
+            setupScreenOptions?.microphoneButton?.isEnabled != false
+    }
+
+    private fun shouldAudioDeviceButtonBeEnabled(
+        callingState: CallingState,
+        setupScreenOptions: CallCompositeSetupScreenOptions?
+    ): Boolean {
+        return setupScreenOptions?.audioDeviceButton?.isEnabled ?: true &&
+            !shouldControlsBeDisabled(callingState)
     }
 
     private fun shouldControlsBeDisabled(callingState: CallingState): Boolean {
         if (callingState.isDisconnected())
             return false
         return callingState.joinCallIsRequested || callingState.callingStatus != CallingStatus.NONE
     }
+
+    private fun callOnClickHandler(
+        context: Context,
+        buttonOptions: CallCompositeButtonOptions?,
+    ) {
+        try {
+            buttonOptions?.onClickHandler?.handle(
+                createButtonClickEvent(context, buttonOptions)
+            )
+        } catch (_: Exception) {
```

don't hide exceptions, log.e

---

Added